sabnzbd/newsunpack.py

```diff
@@ -1081,6 +1070,28 @@ def seven_extract_core(sevenset, extensions, extraction_path, one_folder, delete
             except:
                 logging.warning(T("Deleting %s failed!"), sevenset)
 
+    # 7z unpack had a problem
+    if ret == 2:
+        # Fatal error with 7z/7za, so let's find the reason in the 7z output
+        if "Data Error in encrypted file. Wrong password?" in output:
+            msg = "%s (%s)" % (T("Unpacking failed, archive requires a password"), os.path.basename(sevenset))
+        elif "Disk full." in output or "No space left on device" in output:
+            # note: the above does not work with 7z version 16.02, and does from with 7z 19.00 and higher
+            ret = 1
+            msg = T("Unpacking failed, write error or disk is full?") + " (%s)" % setname_from_path(sevenset)
+        elif "ERROR: CRC Failed" in output:
+            # We can output a more general error
+            ret = 1
+            msg = T('ERROR: CRC failed in "%s"') % setname_from_path(sevenset)
+        else:
+            # Default message
+            msg = T("Could not unpack %s") % setname_from_path(sevenset)
+
+    # 7z cases we don't handle
+    if ret > 2:
+        # 7 = Bad command line parameters, 8 = Not enough memory for operation, 255 = User stopped the process
+        msg = "Unexpected 7z error. Check debug logging"
```

Comment:
Let's include this with the code above to output 1 general message.
Maybe we already have "Check the logs" type message in the translations?

Reply:
I now use two existing messages.